From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tyler Baker <tyler.baker@oss.qualcomm.com>
Date: Fri, 2 Oct 2026 10:02:59 -0400
Subject: [PATCH] package/ffmpeg: bump to version 7.1.2

At the end of the input, the ffmpeg 6.1 command line tool takes every
frame the decoder still returns into its filter graph before it runs the
graph. With a V4L2 mem2mem decoder those frames hold CAPTURE buffers, so
a decoder that needs a CAPTURE buffer back to finish draining waits
forever. On the Qualcomm venus decoder (QCM2290), a 60-frame 1280x720
H.264 stream hangs after V4L2_DEC_CMD_STOP with 58 frames decoded and 18
of the 20 CAPTURE buffers held by ffmpeg. ffmpeg 7.0 runs filtering in
its own thread, which keeps returning the buffers while the decoder
drains.

Use the release tarball, whose signature can be checked. Drop
--disable-crystalhd: 7.0 removed the decoder and configure rejects the
option. Rebase 0001 and 0003; drop 0005, 0006 and 0007, which 7.1.2
already contains.

Signed-off-by: Tyler Baker <tyler.baker@oss.qualcomm.com>
---
 ...-x86-yuv2rgb-Fix-build-without-SSSE3.patch | 12 +++--
 .../0003-libavutil-Fix-mips-build.patch       |  8 +--
 ...0005-avcodec-mmaldec-Fix-build-error.patch | 32 ------------
 ...ay-dvd-Use-correct-pointer-types-on-.patch | 52 -------------------
 ...hwcontext-vaapi-use-the-correct-type.patch | 28 ----------
 package/ffmpeg/ffmpeg.hash                    |  5 +-
 package/ffmpeg/ffmpeg.mk                      |  9 ++--
 7 files changed, 19 insertions(+), 127 deletions(-)
 delete mode 100644 package/ffmpeg/0005-avcodec-mmaldec-Fix-build-error.patch
 delete mode 100644 package/ffmpeg/0006-avcodec-pcm-bluray-dvd-Use-correct-pointer-types-on-.patch
 delete mode 100644 package/ffmpeg/0007-avutil-hwcontext-vaapi-use-the-correct-type.patch

diff --git a/package/ffmpeg/0001-swscale-x86-yuv2rgb-Fix-build-without-SSSE3.patch b/package/ffmpeg/0001-swscale-x86-yuv2rgb-Fix-build-without-SSSE3.patch
index aa56e5a..be40797 100644
--- a/package/ffmpeg/0001-swscale-x86-yuv2rgb-Fix-build-without-SSSE3.patch
+++ b/package/ffmpeg/0001-swscale-x86-yuv2rgb-Fix-build-without-SSSE3.patch
@@ -14,6 +14,8 @@ Signed-off-by: Michael Niedermayer <michael@niedermayer.cc>
 Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>
 [yann.morin.1998@free.fr: fix reference top patchwork and SoB order]
 Signed-off-by: Yann E. MORIN <yann.morin.1998@free.fr>
+[Tyler: rebased for version 7.1.2]
+Signed-off-by: Tyler Baker <tyler.baker@oss.qualcomm.com>
 ---
  libswscale/x86/yuv2rgb.c | 2 ++
  1 file changed, 2 insertions(+)
@@ -22,7 +24,7 @@ diff --git a/libswscale/x86/yuv2rgb.c b/libswscale/x86/yuv2rgb.c
 index c12e88cbb5..4791e5b93a 100644
 --- a/libswscale/x86/yuv2rgb.c
 +++ b/libswscale/x86/yuv2rgb.c
-@@ -70,6 +70,7 @@ av_cold SwsFunc ff_yuv2rgb_init_x86(SwsContext *c)
+@@ -243,6 +243,7 @@ av_cold SwsFunc ff_yuv2rgb_init_x86(SwsContext *c)
  #if HAVE_X86ASM
      int cpu_flags = av_get_cpu_flags();
  
@@ -30,14 +32,14 @@ index c12e88cbb5..4791e5b93a 100644
      if (EXTERNAL_SSSE3(cpu_flags)) {
          switch (c->dstFormat) {
          case AV_PIX_FMT_RGB32:
-@@ -98,6 +99,7 @@ av_cold SwsFunc ff_yuv2rgb_init_x86(SwsContext *c)
-             return yuv420_rgb15_ssse3;
+@@ -275,6 +276,7 @@ av_cold SwsFunc ff_yuv2rgb_init_x86(SwsContext *c)
+ #endif
          }
      }
 +#endif
  
-     if (EXTERNAL_MMXEXT(cpu_flags)) {
-         switch (c->dstFormat) {
+ #endif /* HAVE_X86ASM */
+     return NULL;
 -- 
 2.26.2
 
diff --git a/package/ffmpeg/0003-libavutil-Fix-mips-build.patch b/package/ffmpeg/0003-libavutil-Fix-mips-build.patch
index aeb272c..ea1f912 100644
--- a/package/ffmpeg/0003-libavutil-Fix-mips-build.patch
+++ b/package/ffmpeg/0003-libavutil-Fix-mips-build.patch
@@ -14,6 +14,8 @@ http://ffmpeg.org/pipermail/ffmpeg-devel/2021-June/281272.html
 Signed-off-by: Bernd Kuhls <bernd@kuhls.net>
 [Dario: make the patch to be applied with fuzz factor 0]
 Signed-off-by: Dario Binacchi <dario.binacchi@amarulasolutions.com>
+[Tyler: rebased for version 7.1.2]
+Signed-off-by: Tyler Baker <tyler.baker@oss.qualcomm.com>
 ---
  configure            | 2 ++
  libavutil/mips/cpu.c | 6 +++---
@@ -23,15 +25,15 @@ diff --git a/configure b/configure
 index a89cfa6d959d..6ec00df75312 100755
 --- a/configure
 +++ b/configure
-@@ -2227,6 +2227,7 @@ HEADERS_LIST="
+@@ -2297,6 +2297,7 @@ HEADERS_LIST="
      OpenGL_gl3_h
      poll_h
      pthread_np_h
 +    sys_auxv_h
+     sys_hwprobe_h
      sys_param_h
      sys_resource_h
-     sys_select_h
-@@ -6472,6 +6473,7 @@ check_func_headers VideoToolbox/VTCompressionSession.h VTCompressionSessionPrepa
+@@ -6672,6 +6673,7 @@ check_func_headers VideoToolbox/VTCompressionSession.h VTCompressionSessionPrepa
  check_func_headers VideoToolbox/VideoToolbox.h VTPixelRotationSessionCreate -framework VideoToolbox
  check_headers windows.h
  check_headers asm/types.h
diff --git a/package/ffmpeg/0005-avcodec-mmaldec-Fix-build-error.patch b/package/ffmpeg/0005-avcodec-mmaldec-Fix-build-error.patch
deleted file mode 100644
index b3b9012..0000000
--- a/package/ffmpeg/0005-avcodec-mmaldec-Fix-build-error.patch
+++ /dev/null
@@ -1,32 +0,0 @@
-From e8a49b1424d405441fb23c2850274a84e95bd892 Mon Sep 17 00:00:00 2001
-From: Zhao Zhili <zhilizhao@tencent.com>
-Date: Tue, 21 Nov 2023 04:05:08 +0800
-Subject: [PATCH] avcodec/mmaldec: Fix build error
-
-Fix #10670.
-
-Signed-off-by: Zhao Zhili <zhilizhao@tencent.com>
-
-Upstream: http://git.videolan.org/?p=ffmpeg.git;a=commitdiff;h=e8a49b1424d405441fb23c2850274a84e95bd892
-
-Signed-off-by: Bernd Kuhls <bernd@kuhls.net>
----
- libavcodec/mmaldec.c | 2 +-
- 1 file changed, 1 insertion(+), 1 deletion(-)
-
-diff --git a/libavcodec/mmaldec.c b/libavcodec/mmaldec.c
-index 4bb3eb22ef..2dc7bbee04 100644
---- a/libavcodec/mmaldec.c
-+++ b/libavcodec/mmaldec.c
-@@ -843,7 +843,7 @@ static const AVClass ffmmal_dec_class = {
-         .flush          = ffmmal_flush, \
-         .p.priv_class   = &ffmmal_dec_class, \
-         .p.capabilities = AV_CODEC_CAP_DELAY | AV_CODEC_CAP_HARDWARE, \
--        .caps_internal  = FF_CODEC_CAP_NOT_INIT_THREADSAFE \
-+        .caps_internal  = FF_CODEC_CAP_NOT_INIT_THREADSAFE, \
-         .p.pix_fmts     = (const enum AVPixelFormat[]) { AV_PIX_FMT_MMAL, \
-                                                          AV_PIX_FMT_YUV420P, \
-                                                          AV_PIX_FMT_NONE}, \
--- 
-2.30.2
-
diff --git a/package/ffmpeg/0006-avcodec-pcm-bluray-dvd-Use-correct-pointer-types-on-.patch b/package/ffmpeg/0006-avcodec-pcm-bluray-dvd-Use-correct-pointer-types-on-.patch
deleted file mode 100644
index a77a0cf..0000000
--- a/package/ffmpeg/0006-avcodec-pcm-bluray-dvd-Use-correct-pointer-types-on-.patch
+++ /dev/null
@@ -1,52 +0,0 @@
-From 347a70f101be28f8d78e8fd62ffc3a78324f49e9 Mon Sep 17 00:00:00 2001
-From: Andreas Rheinhardt <andreas.rheinhardt@outlook.com>
-Date: Thu, 28 Mar 2024 05:35:36 +0100
-Subject: [PATCH] avcodec/pcm-bluray/dvd: Use correct pointer types on BE
-
-Signed-off-by: Andreas Rheinhardt <andreas.rheinhardt@outlook.com>
-Signed-off-by: Waldemar Brodkorb <wbx@openadk.org>
-Upstream: https://git.ffmpeg.org/gitweb/ffmpeg.git/commit/347a70f101be28f8d78e8fd62ffc3a78324f49e9
----
- libavcodec/pcm-bluray.c | 5 +++--
- libavcodec/pcm-dvd.c    | 2 +-
- 2 files changed, 4 insertions(+), 3 deletions(-)
-
-diff --git a/libavcodec/pcm-bluray.c b/libavcodec/pcm-bluray.c
-index f65609514a..235020d78f 100644
---- a/libavcodec/pcm-bluray.c
-+++ b/libavcodec/pcm-bluray.c
-@@ -167,7 +167,7 @@ static int pcm_bluray_decode_frame(AVCodecContext *avctx, AVFrame *frame,
-             samples *= num_source_channels;
-             if (AV_SAMPLE_FMT_S16 == avctx->sample_fmt) {
- #if HAVE_BIGENDIAN
--                bytestream2_get_buffer(&gb, dst16, buf_size);
-+                bytestream2_get_buffer(&gb, (uint8_t*)dst16, buf_size);
- #else
-                 do {
-                     *dst16++ = bytestream2_get_be16u(&gb);
-@@ -187,7 +187,8 @@ static int pcm_bluray_decode_frame(AVCodecContext *avctx, AVFrame *frame,
-             if (AV_SAMPLE_FMT_S16 == avctx->sample_fmt) {
-                 do {
- #if HAVE_BIGENDIAN
--                    bytestream2_get_buffer(&gb, dst16, avctx->ch_layout.nb_channels * 2);
-+                    bytestream2_get_buffer(&gb, (uint8_t*)dst16,
-+                                           avctx->ch_layout.nb_channels * 2);
-                     dst16 += avctx->ch_layout.nb_channels;
- #else
-                     channel = avctx->ch_layout.nb_channels;
-diff --git a/libavcodec/pcm-dvd.c b/libavcodec/pcm-dvd.c
-index 419b2a138f..319746c62e 100644
---- a/libavcodec/pcm-dvd.c
-+++ b/libavcodec/pcm-dvd.c
-@@ -157,7 +157,7 @@ static void *pcm_dvd_decode_samples(AVCodecContext *avctx, const uint8_t *src,
-     switch (avctx->bits_per_coded_sample) {
-     case 16: {
- #if HAVE_BIGENDIAN
--        bytestream2_get_buffer(&gb, dst16, blocks * s->block_size);
-+        bytestream2_get_buffer(&gb, (uint8_t*)dst16, blocks * s->block_size);
-         dst16 += blocks * s->block_size / 2;
- #else
-         int samples = blocks * avctx->ch_layout.nb_channels;
--- 
-2.39.2
-
diff --git a/package/ffmpeg/0007-avutil-hwcontext-vaapi-use-the-correct-type.patch b/package/ffmpeg/0007-avutil-hwcontext-vaapi-use-the-correct-type.patch
deleted file mode 100644
index 177b4f1..0000000
--- a/package/ffmpeg/0007-avutil-hwcontext-vaapi-use-the-correct-type.patch
+++ /dev/null
@@ -1,28 +0,0 @@
-From 6f8e365a2af2b6b21701d41eed3b2e3f8a436eeb Mon Sep 17 00:00:00 2001
-From: James Almer <jamrial@gmail.com>
-Date: Wed, 31 Jul 2024 10:00:54 -0300
-Subject: [PATCH] avutil/hwcontext_vaapi: use the correct type for
- VASurfaceAttribExternalBuffers.buffers
-
-Should fix ticket #11115.
-
-Signed-off-by: James Almer <jamrial@gmail.com>
-Upstream: https://git.ffmpeg.org/gitweb/ffmpeg.git/commit/6f8e365a2af2b6b21701d41eed3b2e3f8a436eeb
-Signed-off-by: Thomas Perale <thomas.perale@mind.be>
----
- libavutil/hwcontext_vaapi.c | 2 +-
- 1 file changed, 1 insertion(+), 1 deletion(-)
-
-diff --git a/libavutil/hwcontext_vaapi.c b/libavutil/hwcontext_vaapi.c
-index 4cb25dd03212e..15fd84aa40679 100644
---- a/libavutil/hwcontext_vaapi.c
-+++ b/libavutil/hwcontext_vaapi.c
-@@ -1225,7 +1225,7 @@ static int vaapi_map_from_drm(AVHWFramesContext *src_fc, AVFrame *dst,
- 
-     if (!use_prime2 || vas != VA_STATUS_SUCCESS) {
-         int k;
--        unsigned long buffer_handle;
-+        uintptr_t buffer_handle;
-         VASurfaceAttribExternalBuffers buffer_desc;
-         VASurfaceAttrib buffer_attrs[2] = {
-             {
diff --git a/package/ffmpeg/ffmpeg.hash b/package/ffmpeg/ffmpeg.hash
index 50498a6..9a7854b 100644
--- a/package/ffmpeg/ffmpeg.hash
+++ b/package/ffmpeg/ffmpeg.hash
@@ -1,5 +1,8 @@
+# Locally calculated after checking pgp signature
+# https://ffmpeg.org/releases/ffmpeg-7.1.2.tar.xz.asc
+# with key FCF986EA15E6E293A5644F10B4322F04D67658D8
+sha256  089bc60fb59d6aecc5d994ff530fd0dcb3ee39aa55867849a2bbc4e555f9c304  ffmpeg-7.1.2.tar.xz
 # Locally calculated
-sha256  f0b2b07c7014b7aa8c4689095b83a49a85d41887049fc5302543aec0bef7d066  ffmpeg-n6.1.2-27-ge16ff06adb-git4.tar.gz
 sha256  8177f97513213526df2cf6184d8ff986c675afb514d4e68a404010521b880643  COPYING.GPLv2
 sha256  b634ab5640e258563c536e658cad87080553df6f34f62269a21d554844e58bfe  COPYING.LGPLv2.1
 sha256  cb48bf09a11f5fb576cddb0431c8f5ed0a60157a9ec942adffc13907cbe083f2  LICENSE.md
diff --git a/package/ffmpeg/ffmpeg.mk b/package/ffmpeg/ffmpeg.mk
index 5c00ba2..71fbd31 100644
--- a/package/ffmpeg/ffmpeg.mk
+++ b/package/ffmpeg/ffmpeg.mk
@@ -4,12 +4,10 @@
 #
 ################################################################################
 
-# Update FFMPEG_CPE_ID_VERSION if needed
-FFMPEG_VERSION = n6.1.2-27-ge16ff06adb
-FFMPEG_SITE = https://git.ffmpeg.org/ffmpeg.git
-FFMPEG_SITE_METHOD = git
+FFMPEG_VERSION = 7.1.2
+FFMPEG_SOURCE = ffmpeg-$(FFMPEG_VERSION).tar.xz
+FFMPEG_SITE = https://ffmpeg.org/releases
 FFMPEG_INSTALL_STAGING = YES
-FFMPEG_CPE_ID_VERSION = 6.1.2
 
 FFMPEG_LICENSE = LGPL-2.1+, libjpeg license
 FFMPEG_LICENSE_FILES = LICENSE.md COPYING.LGPLv2.1
@@ -34,7 +32,6 @@ FFMPEG_CONF_OPTS = \
 	--disable-gray \
 	--enable-swscale-alpha \
 	--disable-small \
-	--disable-crystalhd \
 	--disable-dxva2 \
 	--enable-runtime-cpudetect \
 	--disable-hardcoded-tables \
